src/Avalonia.Input/TextInput/InputMethodManager.cs

```diff
@@ -75,27 +89,32 @@ public void SetFocusedElement(IInputElement? element)
             if(_focusedElement == element)
                 return;
             _focusedElement = element;
-            
-            var inputMethod = (element?.VisualRoot as ITextInputMethodRoot)?.InputMethod;
+
+            var root = element?.VisualRoot as ITextInputMethodRoot;
+
+            NotifyInputMethodUpdated(root);
+        }
+
+        public void NotifyInputMethodUpdated(ITextInputMethodRoot? root)
+        {
+            // TODO how to filter the window with the current focused element?
```

Comment:
`_focusedElement?.VisualRoot == root`

Reply:
thanks.
Just about time. Tested with multi-window mode FVim and working.